modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcBatchSelfTest.java

```diff
@@ -636,6 +619,61 @@ public void testBatchClearPrepared() throws SQLException {
         assertEquals(0L, personsCount(), "Test table should be empty after empty batch is performed.");
     }
 
+    @Test
+    public void testTransactionalBatch() throws Exception {
+        conn.setAutoCommit(false);
+
+        int rowsCnt = 5;
+
+        Callable<int[]> batchOp = () -> {
+            for (int i = 0; i < rowsCnt; i++) {
+                stmt.addBatch("insert into Person (id, firstName, lastName, age) values " + valuesRow(i));
+            }
+
+            return stmt.executeBatch();
+        };
+
+        // Check rollback.
+        checkBatchTxResult(rowsCnt, batchOp, false);
+
+        // Check commit.
+        checkBatchTxResult(rowsCnt, batchOp, true);
+    }
+
+    @Test
+    public void testTransactionalBatchPrepared() throws Exception {
+        conn.setAutoCommit(false);
+
+        int rowsCnt = 5;
+
+        Callable<int[]> batchOp = () -> {
+            populatePrepared(rowsCnt);
+
+            return pstmt.executeBatch();
```

Comment:
do you have test for transaction spanning several calls to `executeBatch`?

Reply:
This test case is now covered by integration tests.